common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -431,6 +431,13 @@ private S getExtensionInstance(ExtensionDefinition<S> definition, ClassLoader lo
             if (definition == null) {
                 throw new EnhancedServiceNotFoundException("not found service provider for : " + type.getName());
             }
+            int state;
+            if (BitUtils.isSetBit(state = definition.getState(), Constants.DISABLE_EXTENSION_DEFINITION)) {
```

Comment:
AS  I understand  the  `ExtensionDefinition state` is a binary state (enable or disable), so it can be represented as a boolean. Why introduce complicated bit operations? It's too complicated to go through multiple steps of AND, OR, and XOR, so why not use CAS for boolean comparisons? Please explain what more considerations are here.

----

按照我的理解state是一个二元状态（enable or disable），因此，可以使用boolean类型来表示。为什么要引入复杂的位运算。经过与、或和异或的多步运算逻辑显得太复杂，为什么不使用CAS做boolean值比较？请说明这里引人位运算更多的考虑是什么？

Reply:
I used to think that there are two states of availability and unavailability when it is running. In addition to the state of runtime, I thought there was a non-running judgment. Now that I think about it, there is no need to store redundant data in memory when it is not running.
-----------
我之前以为他运行时存在可用与不可用俩种状态，除了运行时的状态以外，我以为还有非运行时的判断。现在思考了一下非运行时也没必要保存在内存中做冗余数据的存储了。